From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ronald Claveau <linux-kernel-dev@aliel.fr>
Date: Fri, 3 Apr 2026 12:11:34 +0200
Subject: usb: host: xhci-plat: Add support for Amlogic T7 SoC

The Amlogic T7 SoC integrates Corigine USB IP
which is xHCI compliant.
Add a platform data entry with the required quirks:
XHCI_NO_64BIT_SUPPORT because the controller incorrectly
advertises 64-bit DMA capability,
and XHCI_RESET_ON_RESUME.

Signed-off-by: Ronald Claveau <linux-kernel-dev@aliel.fr>
---
 drivers/usb/host/xhci-plat.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/usb/host/xhci-plat.c b/drivers/usb/host/xhci-plat.c
index 111111111111..222222222222 100644
--- a/drivers/usb/host/xhci-plat.c
+++ b/drivers/usb/host/xhci-plat.c
@@ -124,6 +124,10 @@ static const struct xhci_plat_priv xhci_plat_brcm = {
 	.quirks = XHCI_RESET_ON_RESUME | XHCI_SUSPEND_RESUME_CLKS,
 };
 
+static const struct xhci_plat_priv xhci_plat_amlogic_t7 = {
+	.quirks = XHCI_NO_64BIT_SUPPORT | XHCI_RESET_ON_RESUME,
+};
+
 static const struct of_device_id usb_xhci_of_match[] = {
 	{
 		.compatible = "generic-xhci",
@@ -147,6 +151,9 @@ static const struct of_device_id usb_xhci_of_match[] = {
 	}, {
 		.compatible = "brcm,bcm7445-xhci",
 		.data = &xhci_plat_brcm,
+	}, {
+		.compatible = "amlogic,t7-xhci",
+		.data = &xhci_plat_amlogic_t7,
 	},
 	{},
 };
-- 
Armbian

